frontend/packages/pipelines-plugin/src/components/pipelines/modals/common/PipelineWorkspacesSection.tsx

```diff
@@ -83,26 +84,33 @@ const PipelineWorkspacesSection: React.FC = () => {
   return (
     workspaces.length > 0 && (
       <FormSection title={t('pipelines-plugin~Workspaces')} fullWidth>
-        {workspaces.map((workspace, index) => (
-          <div className="form-group" key={workspace.name}>
-            <DropdownField
-              name={`workspaces.${index}.type`}
-              label={workspace.name}
-              items={volumeTypeOptions}
-              onChange={(type) =>
-                setFieldValue(
-                  `workspaces.${index}.data`,
-                  type === VolumeTypes.EmptyDirectory ? { emptyDir: {} } : {},
-                  // Validation is automatically done by DropdownField useFormikValidationFix
-                  false,
-                )
-              }
-              fullWidth
-              required={!workspace.optional}
-            />
-            {getVolumeTypeFields(workspace.type, index, t)}
-          </div>
-        ))}
+        {workspaces.map((workspace, index) => {
+          const items = { ...volumeTypeOptions };
+          if (!workspace.optional) {
```

Comment:
Just a query is it better to delete the key or obtain 
` {[volumeTypeOptions.NoWorkspace],...items} = volumeTypeOptions `

Reply:
Well if you're safe and know what you're doing, delete isn't a problem. But you have made a good call. I'll update it to an omit.